### PR TITLE
Model code generation

### PR DESCRIPTION
**Pull request by nickfloros**
The swagger specification defines the models required by various transactions. For the project I am working we have rather complex structures and lots of enumerations. Hence found it useful to have in code the models defined in the swagger specification. 

In my attempt to keep my additions clear of your code all my new code is [lib/codegen-model.js](https://github.com/nickfloros/swagger-js-codegen/blob/master/lib/codegen-model.js) . However this resulted in adding [lib/index.js](https://github.com/nickfloros/swagger-js-codegen/blob/master/lib/index..js) as the entry point to the packages.

Unit tests pass with the exception of node-api.js where it breaks because it aborts due to warnings ... I do not understand what is happening ... 

I am also in the process of updating grunt-swagger-js-codegen package.

### CODE REVIEW

**Comment by wcandillon** [comment]
@nickfloros can you fix the conflicts? It should be trivial.

**Comment by wcandillon** [comment]
@nickfloros also a rebase of the commits would be nice :)

File: tests/node-api.js [hunk 1]
```diff
@@ -11,8 +11,11 @@ vows.describe('Test Generated API').addBatch({
     'Test Generated code for the 28.io Auth API': {
         topic: function(){
             var swagger = JSON.parse(fs.readFileSync('tests/apis/auth.json', 'UTF-8'));
+            var c = CodeGen.getNodeCode({ className: 'Auth', swagger: swagger });
+            console.log(c);
```
Comment (wcandillon) [inline tests/node-api.js]: @nickfloros can you remove the `console.log`?

**Comment by wcandillon** [comment]
@nickfloros Could show me an model that shows the power of these model in action? If it could be integrated into the test suite that would be even better.

File: tests/models/model-1.json.orig [hunk 2]
```diff
@@ -0,0 +1,201 @@
+{
```
Comment (nickfloros) [inline tests/models/model-1.json.orig]: Thanks I have more changes to check in. Struggling with the rebate, I think I will take fresh copy from you and try again.

**Comment by mattapperson** [comment]
@nickfloros are you still working on this?

**Comment by nickfloros** [comment]
Yes, but have been very busy with work.

Can you please reject my pull request, and I will merge with the latest
version.

Thanks

Nick

**Comment by nickfloros** [comment]
Hi

 Created a new pull request but ...

 if you find all too complex please do not hesitate to reject it.

 Thanks

Nick

On Sat, Feb 21, 2015 at 7:08 AM, William Candillon <notifications@github.com

**Pull request closed without merging**